src/plugins/intel_cpu/src/serialize.cpp

```diff
@@ -4,69 +4,43 @@
 #include "serialize.h"
 
 #include <openvino/pass/serialize.hpp>
-
 #include <pugixml.hpp>
+#include <transformations/utils/utils.hpp>
 
 using namespace InferenceEngine;
 
 namespace ov {
 namespace intel_cpu {
 namespace {
-    std::string to_string(InferenceEngine::Layout layout) {
-        std::stringstream ss;
-        ss << layout;
-        return ss.str();
-    }
 
-    InferenceEngine::Layout layout_from_string(const std::string & name) {
-        static const std::unordered_map<std::string, InferenceEngine::Layout> layouts = {
-            { "ANY", InferenceEngine::Layout::ANY },
-            { "NCHW", InferenceEngine::Layout::NCHW },
-            { "NHWC", InferenceEngine::Layout::NHWC },
-            { "NCDHW", InferenceEngine::Layout::NCDHW },
-            { "NDHWC", InferenceEngine::Layout::NDHWC },
-            { "OIHW", InferenceEngine::Layout::OIHW },
-            { "C", InferenceEngine::Layout::C },
-            { "CHW", InferenceEngine::Layout::CHW },
-            { "HWC", InferenceEngine::Layout::HWC },
-            { "HW", InferenceEngine::Layout::HW },
-            { "NC", InferenceEngine::Layout::NC },
-            { "CN", InferenceEngine::Layout::CN },
-            { "BLOCKED", InferenceEngine::Layout::BLOCKED }
-        };
-        auto it = layouts.find(name);
-        if (it != layouts.end()) {
-            return it->second;
-        }
-        IE_THROW(NetworkNotRead) << "Unknown layout with name '" << name << "'";
-    }
-
-    template <typename T>
-    void setInfo(pugi::xml_object_range<pugi::xml_named_node_iterator>&& nodes, T&& info) {
-        auto nodes_it = nodes.begin();
-        auto info_iter = info.begin();
-        for (; nodes_it != nodes.end(); ++nodes_it, ++info_iter) {
-            auto name_attr = nodes_it->attribute("name");
-            auto precision_attr = nodes_it->attribute("precision");
-            auto layout_attr = nodes_it->attribute("layout");
-
-            if (!name_attr || !precision_attr || !layout_attr || info_iter == info.end()) {
-                IE_THROW(NetworkNotRead) << "The inputs/outputs information is invalid.";
-            }
-
-            info_iter->second->setName(name_attr.value());
-            info_iter->second->setPrecision(Precision::FromStr(precision_attr.value()));
-            info_iter->second->setLayout(layout_from_string(layout_attr.value()));
+template <typename T>
+void setInfo(pugi::xml_object_range<pugi::xml_named_node_iterator>&& nodes, T&& info) {
```

Comment:
do we need it know? I suppose it was required only for API 1.0 where blob and input precisions / layouts can mismatch. In API 2.0 they are aligned

Reply:
It was used for the precision mismatch between original model and compiled model (transformation issues), if this problem has been fixed we can remove it.